Assets/Scripts/Base/BaseSpawner.cs

```diff
@@ -14,24 +14,30 @@ public class BaseSpawner : NetworkBehaviour
 
     #region Server
     [Server]
-    private GameObject SpawnArmy(Unit[] units)
+    private GameObject SpawnArmy(IdentityInfo identity, int count)
     {
         GameObject armyInstance = Instantiate(
             armyPrefab,
             armySpawnPoint.position,
             armySpawnPoint.rotation);
-        IdentityInfo baseIdentity = GetComponent<ObjectIdentity>().Identity;  // TODO: move to OnServerStart() if base identity doesn't change and hope race condition doesn't happen
-        armyInstance.GetComponent<ObjectIdentity>().SetIdentity(baseIdentity);
+        armyInstance.GetComponent<ObjectIdentity>().SetIdentity(identity);
+
         Army army = armyInstance.GetComponent<Army>();
-        army.SetUnits(units);
+        army.SetUnits(identity, count);
         NetworkServer.Spawn(armyInstance, connectionToClient);
         return armyInstance;
-    } 
+    }
 
+    /// <summary>
+    /// Spawns an army with <count> number of units with <identity> and moves it to <position>
+    /// </summary>
+    /// <param name="identity"> IdentityInfo of each unit. This mean every unit in the army will start with an identical unit</param>
+    /// <param name="count"> Number of units the army will spawn with </param>
+    /// <param name="position">Target movement position </param>
     [Command]
-    public void CmdSpawnMoveArmy(Unit[] units, Vector3 position)
+    public void CmdSpawnMoveArmy(IdentityInfo identity, int count, Vector3 position)
```

Comment:
Try calling `army.CmdMove` in here without the need of `army.TryMove` (server function calling another server function)?